rest-api-spec/src/main/resources/rest-api-spec/test/indices.get_settings/40_number_of_routing_shards.yml

```
@@ -0,0 +1,41 @@
+---
+setup:
+  - do:
+      indices.create:
+        body:
+          settings:
+            index:
+              number_of_routing_shards: 4
+              number_of_shards: 2
+              number_of_replicas: 1
+        index: test-index
+
+  - do:
+      indices.create:
+        body:
+          settings:
+            index:
+              number_of_shards: 2
+              number_of_replicas: 1
+        index: test-index1
+
```

All the YAML tests are faiiing.  Looking at other tests you might need this (purely a guess):
```suggestion
  - do:
      cluster.health:
        wait_for_status: green
```

---

The failure is caused by this integration test which crashes the cluster: https://github.com/opensearch-project/OpenSearch/blob/20536eef239031af1161fa341a57597e7219fcd5/client/rest-high-level/src/test/java/org/opensearch/client/IndicesClientIT.java#L978, the reason is that now the index setting `number_of_routing_shards` exists in IndexMetadata if it was explicitly set on index creation, but when calling shrink/split/clone API, that setting cannot be copied from the source index, so the setting `number_of_routing_shards` should be `NotCopyableOnResize`, so I add this property to the definition of the setting..